lib/grammers-client/src/types/inline_query.rs

```diff
@@ -0,0 +1,169 @@
+use crate::{client::Client, utils::generate_random_id, InputMessage};
+use grammers_mtsender::InvocationError;
+use grammers_tl_types as tl;
+
+/// Represents an inline query update, which occurs when you sign in as a bot and a user sends an
+/// inline query such as `@bot query`.
+pub struct InlineQuery {
+    query: tl::types::UpdateBotInlineQuery,
+    client: Client,
+}
+
+/// An inline query answer builder.
+pub struct Answer {
+    request: tl::functions::messages::SetInlineBotResults,
+    client: Client,
+}
+
+impl InlineQuery {
+    pub(crate) fn new(client: &Client, query: tl::types::UpdateBotInlineQuery) -> Self {
+        Self {
+            query,
+            client: client.clone(),
+        }
+    }
+
+    // User that sent the query.
+    pub fn user_id(&self) -> i32 {
```

Comment:
Can't this return the entire `Chat`? Is the user not in the entity map?

Reply:
I don't now if inline queries are bind to a specific chat, however I could only find `user_id` [here](https://core.telegram.org/constructor/updateBotInlineQuery).

Reply:
Individual updates themselves don't include the chats, because a single batch may share the same chat for many updates, so it would be wasteful. Instead, the `Updates` constructor is the one with the chats, which for the time being, is an `Arc<ChatMap>`:

https://github.com/Lonami/grammers/blob/f2106e8e234b021819b42e2244d111c72c6279d0/lib/grammers-client/src/types/callback_query.rs#L36-L46

We probably can do something similar here (Telegram *should* send the user). Regardless, we should always aim to return a `Chat`, for consistency, even if it really can only have the `id` field.

Reply:
Done.